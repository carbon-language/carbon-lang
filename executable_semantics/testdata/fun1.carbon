// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
// Exceptions. See /LICENSE for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

package ExecutableSemanticsTest api;

fn f(x: i32) -> i32 {
  return x - 1;
}

fn main() -> i32 {
  return f(1);
}
