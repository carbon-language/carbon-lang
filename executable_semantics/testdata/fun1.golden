********** source program **********
fn f Int: x -> Int {
return (x - 1);

}
fn main () -> Int {
return f(0 = 1);

}
********** type checking **********
--- step exp Int --->
--- step exp Int --->
--- step exp Int --->
--- step exp Int --->
--- step exp Int --->
--- step exp Int --->
10: type error in call
expected: Int
actual: Tuple(0 = Int)
